packages/fast-usdc-contract/src/exos/advancer.ts

```diff
@@ -239,24 +270,74 @@ export const prepareAdvancerKit = (
           result: undefined,
           ctx: AdvancerVowCtx & { tmpSeat: ZCFSeat },
         ) {
-          const { poolAccount, intermediateRecipient, settlementAddress } =
-            this.state;
-          const { destination, advanceAmount, tmpSeat, ...detail } = ctx;
-          tmpSeat.exit();
-          const amount = harden({
-            denom: usdc.denom,
-            value: advanceAmount.value,
-          });
-          const transferOrSendV =
-            destination.chainId === settlementAddress.chainId
-              ? E(poolAccount).send(destination, amount)
-              : E(poolAccount).transfer(destination, amount, {
-                  forwardOpts: { intermediateRecipient },
-                });
-          return watch(transferOrSendV, this.facets.transferHandler, {
-            destination,
-            advanceAmount,
-            ...detail,
+          return asVow(async () => {
+            const {
+              poolAccount,
+              intermediateRecipientAccount,
+              intermediateRecipientAddress,
+              settlementAddress,
+            } = this.state;
+            const { destination, advanceAmount, tmpSeat, ...detail } = ctx;
+            tmpSeat.exit();
+            const amount = harden({
+              denom: usdc.denom,
+              value: advanceAmount.value,
+            });
+
+            /** @type {AccountId} */
+            const accountId = parseAccountIdArg(destination);
+            const chainId = accountId.reference;
+            const info = await when(chainHub.getChainInfoByChainId(chainId));
+
+            let transferOrSendV;
+            if (chainId === settlementAddress.chainId) {
+              // send to recipient on Agoric
+
+              /** @type {CosmosChainAddress} */
+              const cosmosAddress = makeCosmosAccountId(
+                chainId,
+                'bech32',
+                accountId.accountAddress,
+              );
```

Comment:
Let's move some of this logic into `handleTransactionEvent` where we already take an EUD and make a `chainAddress`: https://github.com/Agoric/agoric-sdk/blob/33a57081edfa8bb33e2b61bb96528cf98ed07e34/packages/fast-usdc-contract/src/exos/advancer.ts#L185

- if `parseAccountIdArg` or `getChainInfoByChainId` throw we should `.skipAdvance()`
- if we get an ecosystem that's unknown or not yet supported we should `.skipAdvance()`

Reply:
done